pandas/_libs/tslib.pyx

```diff
@@ -409,7 +409,7 @@ cpdef array_with_unit_to_datetime(ndarray values, unit, errors='coerce'):
         for i in range(n):
             val = values[i]
```

Comment:
these array_* should be moved to conversion I think (obviously new PR), then tslib.pyx is moot?

Reply:
Let's hold off on that.  I'd like to do auditing before moving, and the array_* funcs may need some work/thought (#17697)